discretisedfield/tests/test_field.py

```diff
@@ -3160,14 +3160,18 @@ def test_mpl(test_field):
     plt.close("all")
 
 
-def test_mpl_dimension(valid_mesh):
-    field = df.Field(valid_mesh, nvdim=1)
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
 
     if valid_mesh.region.ndim != 2:
         with pytest.raises(RuntimeError):
-            field.mpl.scalar()
+            field.mpl()
+    elif nvdim not in [1, 2]:
+        with pytest.raises(ValueError):
+            field.mpl()
```

Comment:
Unfortunately, this is much more complicated: we can plot higher-dimensional vectors if `field.vdim_mapping` is defined. This `dict` maps `vdims` to `dims`. Internally, we swap keys and values to find `vdims` that point along the two `dims` of our field.mesh (the axes of our plot). Additionally, we use the first (or maybe one random?) additional component for colour - with the idea of the implementation of just having 3 vdims, and using colour for the single out-of-plane component (the logic is probably from before our changes). We should consider raising some sort of error (maybe `RunTime`) when nvdim > 3. Thoughts?

Reply:
`mpl()` is mainly meant to be for convenience. However, when `nvdims>3` there is no unique choice for the scalar coloring. Hence, I agree with changing the implementation to raise a `RuntimeError`.